samples/hosting/generic-host-multihosting/Core_7/GenericHost/Extensions.cs

```diff
@@ -0,0 +1,354 @@
+using System.Collections.Concurrent;
+using NServiceBus.Configuration.AdvancedExtensibility;
+
+namespace NServiceBus
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Extensions.DependencyInjection;
+    using Microsoft.Extensions.Hosting;
+    using NServiceBus.ObjectBuilder;
+
+    /// <summary>
+    /// Extension methods to configure NServiceBus for the .NET Core generic host.
+    /// </summary>
+    public static class HostBuilderExtensions
+    {
+        /// <summary>
+        /// Configures the host to start an NServiceBus endpoint.
+        /// </summary>
+        public static IHostBuilder UseMultipleNServiceBus(this IHostBuilder hostBuilder, Func<HostBuilderContext, EndpointConfiguration> endpointConfigurationBuilder)
+        {
+            hostBuilder.ConfigureServices((ctx, serviceCollection) =>
+            {
+                var endpointConfiguration = endpointConfigurationBuilder(ctx);
+                var endpointName = endpointConfiguration.GetSettings().Get<string>("NServiceBus.Routing.EndpointName");
+                var childCollection = new ServiceCollection();
```

Comment:
nitpick but I'd avoid the word `childContainer` as long as there is no actual connection to the generic host's "main ServiceCollection"